src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2317,12 +2317,21 @@ public static void TimeZoneInfo_DisplayNameStartsWithOffset()
                     {
                         string offset = Regex.Match(tzi.DisplayName, @"(-|)[0-9]{2}:[0-9]{2}").Value;
                         TimeSpan ts = TimeSpan.Parse(offset);
-                        if (tzi.BaseUtcOffset != ts && tzi.Id.IndexOf("Morocco", StringComparison.Ordinal) >= 0)
+                        if (tzi.BaseUtcOffset != ts && (tzi.Id.IndexOf("Morocco", StringComparison.Ordinal) >= 0 || tzi.Id.IndexOf("Volgograd", StringComparison.Ordinal) >= 0))
                         {
                             // Windows data can report display name with UTC+01:00 offset which is not matching the actual BaseUtcOffset.
                             // We special case this in the test to avoid the test failures like:
                             //      01:00 != 00:00:00, dn:(UTC+01:00) Casablanca, sn:Morocco Standard Time
-                            Assert.True(tzi.BaseUtcOffset == new TimeSpan(0, 0, 0), $"{offset} != {tzi.BaseUtcOffset}, dn:{tzi.DisplayName}, sn:{tzi.StandardName}");
+                            //      04:00 != 03:00:00, dn:(UTC+04:00) Volgograd, sn:Volgograd Standard Time
+                            if (tzi.Id.IndexOf("Morocco", StringComparison.Ordinal) >= 0)
+                            {
+                                Assert.True(tzi.BaseUtcOffset == new TimeSpan(0, 0, 0), $"{offset} != {tzi.BaseUtcOffset}, dn:{tzi.DisplayName}, sn:{tzi.StandardName}");
+                            }
+                            else
+                            {
+                                // Volgograd, Russia
+                                Assert.True(tzi.BaseUtcOffset == new TimeSpan(3, 0, 0), $"{offset} != {tzi.BaseUtcOffset}, dn:{tzi.DisplayName}, sn:{tzi.StandardName}");
+                            }
                         }
                         else
                         {
```

Comment:
Not part of your change, but just curious why the assert below has `tzi.BaseUtcOffset == ts || tzi.GetUtcOffset(DateTime.Now) == ts`. Why would `tzi.BaseUtcOffset == ts` not always be true?

Reply:
`tzi.BaseUtcOffset` is the UTC offset of the zone when we are in the standard time. If the zone is in daylight saving time, the condition `tzi.BaseUtcOffset == ts` can be false. That is why we add the `tzi.GetUtcOffset(DateTime.Now) == ts` condition. We include the 2 conditions because sometimes the Windows update can delay on a machine during the period of the Standard<->daylight transition, but Windows display names are not updated yet.

Reply:
Ah. Don't reset the PR for it but that might be worth a comment at point.